.circleci/config.yml

```diff
@@ -2010,14 +2066,32 @@ jobs:
             # as a temporaty solution explicitly specify ffi version.
             sudo gem install --no-document ffi:1.12.2 fpm
 
-      - run:
-          name: Bump Agent Version
-          command: |
-            # Update agent version to avoid version conflict with current scalyr repository package.
-            # This way we can also correctly assert that the upgrade has worked - if we didn't do
-            # that, from and to version would be the same
-            echo "$(cat VERSION).dev1-$(git rev-parse --short HEAD)" > VERSION
-            cat VERSION
+#      - run:
+#          name: Bump Agent Version
+#          command: |
+#            # Update agent version to avoid version conflict with current scalyr repository package.
+#            # This way we can also correctly assert that the upgrade has worked - if we didn't do
+#            # that, from and to version would be the same
+#            echo "$(cat VERSION).dev1-$(git rev-parse --short HEAD)" > VERSION
+#            cat VERSION
+
+      - when:
+          condition: << pipeline.parameters.agent_version >>
+          steps:
+            - run:
+                name: Change package version to << pipeline.parameters.agent_version >>.
+                shell: bash
+                command: |
+                  echo << pipeline.parameters.agent_version >> > VERSION
+            - unless:
```

Comment:
Hm, will the second step ever get executed?

Since in the first condition we check if ``agent_version`` is truthy / defined and only if it is, we try to run the first and the second step.

But the second step only runs if ``agent_version`` is falsy / not defined (``unless``) and this will never be true because of the conditional in the parent ``condition`` clause?

Or am I missing / misunderstanding something?

Reply:
Wow, yes, thank you for noticing that. It was a copy-paste from the other job, and I missed that the editor made a wrong indent. Actually, I did not expect that circleci would accept such config as valid.

Reply:
No problem and I believe this would have been caught sooner or later by package tests (we verify that the upgrade works by grepping for a specific git revision in the package version).